client/my-sites/stats/all-time/index.jsx

```diff
@@ -78,3 +97,27 @@ export default React.createClass( {
 		);
 	}
 } );
+
+export default connect( ( state ) => {
+	const siteId = getSelectedSiteId( state );
+	const query = {};
+	const allTimeData = getSiteStatsForQuery( state, siteId, 'stats', query ) || { stats: {} };
+	const {
+		posts,
+		views,
+		visitors,
+		views_best_day,
+		views_best_day_total
+	} = allTimeData.stats;
+
+	return {
+		requesting: isRequestingSiteStatsForQuery( state, siteId, 'stats', query ),
+		query,
+		siteId,
+		posts,
+		views,
+		visitors,
+		views_best_day,
+		views_best_day_total
```

Comment:
Looking at #6049 which is very similar, I think my preferred solution would be to fix the names at the Redux selector level, then use the spread operator.  Though it could certainly be argued that we should leave the property names as-is for simplicity's sake.